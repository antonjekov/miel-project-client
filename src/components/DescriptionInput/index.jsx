import React from 'react'
import styles from './index.module.css';
import { Form } from 'react-bootstrap';

function DescriptionInput({ values, errors, handleChange }) {
    return (
        <Form.Group >
            <Form.Label>Description</Form.Label>
            <Form.Control className={styles['form-control']} type="text" placeholder="Description" value={values.description} name='description' onChange={handleChange} isInvalid={!!errors.description} isValid={values.description && !errors.description}>
            </Form.Control>
            <Form.Control.Feedback type='invalid' >{errors.description}</Form.Control.Feedback>
            <Form.Control.Feedback type='valid'>Look's good</Form.Control.Feedback>
        </Form.Group>
    )
}

export default DescriptionInput