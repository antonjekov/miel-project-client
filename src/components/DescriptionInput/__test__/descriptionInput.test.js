import React from 'react'
import renderer from "react-test-renderer"
import {render, cleanup, screen, fireEvent} from "@testing-library/react"
import DescriptionInput from '..'

describe('DescriptionInput', ()=>{

    afterEach(cleanup)
    it('matches snapshot without error', () => {
        const values = {description: 'some description'}
        const errors = {}
        const tree = renderer.create( <DescriptionInput values={values} errors={errors} handleChange={()=>{}}/>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    })
    
    it('matches snapshot with error', () => {
        const values = {description: 'some description'}
        const errors = {description: 'Some error'}
        const tree = renderer.create( <DescriptionInput values={values} errors={errors} handleChange={()=>{}}/>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    })

})
