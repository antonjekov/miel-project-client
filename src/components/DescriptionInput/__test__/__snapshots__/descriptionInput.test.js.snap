// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionInput matches snapshot with error 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    Description
  </label>
  <input
    className="form-control form-control is-invalid"
    name="description"
    onChange={[Function]}
    placeholder="Description"
    type="text"
    value="some description"
  />
  <div
    className="invalid-feedback"
  >
    Some error
  </div>
  <div
    className="valid-feedback"
  >
    Look's good
  </div>
</div>
`;

exports[`DescriptionInput matches snapshot without error 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    Description
  </label>
  <input
    className="form-control form-control is-valid"
    name="description"
    onChange={[Function]}
    placeholder="Description"
    type="text"
    value="some description"
  />
  <div
    className="invalid-feedback"
  />
  <div
    className="valid-feedback"
  >
    Look's good
  </div>
</div>
`;
